fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/BinaryPredicateStatisticCalculator.java

```diff
@@ -41,125 +46,106 @@ public static Statistics estimateColumnToConstantComparison(Optional<ColumnRefOp
 
     private static Statistics estimateColumnEqualToConstant(Optional<ColumnRefOperator> columnRefOperator,
                                                             ColumnStatistic columnStatistic,
-                                                            OptionalDouble constant,
+                                                            ConstantOperator constant,
                                                             Statistics statistics) {
-        StatisticRangeValues predicateRange;
-        if (constant.isPresent()) {
-            predicateRange = new StatisticRangeValues(constant.getAsDouble(), constant.getAsDouble(), 1);
-        } else {
-            predicateRange = new StatisticRangeValues(NEGATIVE_INFINITY, POSITIVE_INFINITY, 1);
-        }
-
-        Statistics estimatePredicateRange =
-                estimatePredicateRange(columnRefOperator, columnStatistic, predicateRange, statistics);
 
-        Map<Double, Long> histogramTopN;
+        double min = StatisticUtils.convertStatisticsToDouble(constant.getType(), constant.toString())
+                .orElse(NEGATIVE_INFINITY);
+        double max = StatisticUtils.convertStatisticsToDouble(constant.getType(), constant.toString())
+                .orElse(POSITIVE_INFINITY);
+
+        ColumnStatistic estimatedColumnStatistic = ColumnStatistic.builder()
+                .setAverageRowSize(columnStatistic.getAverageRowSize())
+                .setNullsFraction(0)
+                .setMinValue(min)
+                .setMaxValue(max)
+                .setDistinctValuesCount(1)
+                .setType(columnStatistic.getType())
+                .build();
+
+        double rowCount;
         if (columnStatistic.getHistogram() != null) {
-            histogramTopN = columnStatistic.getHistogram().getMCV();
-
-            if (histogramTopN.containsKey(constant.getAsDouble())) {
-                return Statistics.buildFrom(estimatePredicateRange)
-                        .setOutputRowCount(histogramTopN.get(constant.getAsDouble())).build();
+            Map<String, Long> histogramTopN = columnStatistic.getHistogram().getMCV();
```

Comment:
May be we need a new class HistogramStatisticsCalculator?

Reply:
I don't think this is necessary for the time being, because the statistical information calculation of the histogram is only used in BinaryPredicate. I think it is not too late to extract it if it needs to be abstracted in the future.